src/Patch/Node.js

```diff
@@ -0,0 +1,160 @@
+import Native from './Native';
+import CustomElementInternals from '../CustomElementInternals';
+import * as Utilities from '../Utilities';
+
+/**
+ * @param {!CustomElementInternals} internals
+ */
+export default function(internals) {
+  // `Node#nodeValue` is implemented on `Attr`.
+  // `Node#textContent` is implemented on `Attr`, `Element`.
+
+  Utilities.setPropertyUnchecked(Node.prototype, 'insertBefore',
+    /**
+     * @this {Node}
+     * @param {!Node} node
+     * @param {?Node} refNode
+     * @return {!Node}
+     */
+    function(node, refNode) {
+      if (node instanceof DocumentFragment) {
+        const insertedNodes = Array.prototype.slice.apply(node.childNodes);
+        const nativeResult = Native.Node_insertBefore.call(this, node, refNode);
+
+        // DocumentFragments can't be connected, so `disconnectTree` will never
+        // need to be called on a DocumentFragment's children after inserting it.
+
+        if (Utilities.isConnected(this)) {
+          for (let i = 0; i < insertedNodes.length; i++) {
+            internals.connectTree(insertedNodes[i]);
+          }
+        }
+
+        return nativeResult;
+      }
+
+      const nodeWasConnected = Utilities.isConnected(node);
+      const nativeResult = Native.Node_insertBefore.call(this, node, refNode);
+
+      if (nodeWasConnected) {
+        internals.disconnectTree(node);
+      }
+
+      if (Utilities.isConnected(this)) {
+        internals.connectTree(node);
+      }
+
+      return nativeResult;
+    });
+
+  Utilities.setPropertyUnchecked(Node.prototype, 'appendChild',
+    /**
+     * @this {Node}
+     * @param {!Node} node
+     * @return {!Node}
+     */
+    function(node) {
```

Comment:
Suggest collapsing the implementation of this with `insertBefore` since append is equivalent to `insertBefore` with no ref node.